core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/BaseDefaultHeartbeatPropertyProvider.java

```diff
@@ -0,0 +1,162 @@
+   * us identify instability in the analytics backend.
+   */
+  private static UUID uniqueProcessId;
+
+  /**
+   * Name of this provider.
+   */
+  private final String name = "Base";
+
+  public BaseDefaultHeartbeatPropertyProvider() {
+    defaultFields = new HashSet<>();
+    initializeDefaultFields(defaultFields);
+  }
+
+  @Override
+  public String getName() {
+    return this.name;
+  }
+
+  @Override
+  public boolean isKeyWord(String keyword) {
+    return defaultFields.contains(keyword);
+  }
+
+  @Override
+  public Callable<Boolean> setDefaultPayload(final List<String> disableFields,
+      final HeartBeatProviderInterface provider) {
+    return new Callable<Boolean>() {
+
+      // using volatile here to avoid caching in threads.
+      volatile boolean hasSetValues = false;
+      volatile Set<String> enabledProperties = MiscUtils.except(defaultFields, disableFields);
+      @Override
+      public Boolean call() {
+        for (String fieldName : enabledProperties) {
+          try {
+            switch (fieldName) {
+              case "jdkVersion":
+                provider.addHeartBeatProperty(fieldName, getJdkVersion(), true);
+                hasSetValues = true;
+                break;
+              case "sdk-version":
+                provider.addHeartBeatProperty(fieldName, getSdkVersion(), true);
+                hasSetValues = true;
+                break;
+              case "osType":
+                provider.addHeartBeatProperty(fieldName, getOsType(), true);
+                hasSetValues = true;
+                break;
+              case "processSessionId":
+                provider.addHeartBeatProperty(fieldName, getProcessSessionId(), true);
+                hasSetValues = true;
+                break;
+              default:
+                //We won't accept unknown properties in default providers.
+                InternalLogger.INSTANCE.trace("Encountered unknown default property");
+                break;
+            }
+          }
+          catch (Exception e) {
+           InternalLogger.INSTANCE.warn("Failed to obtain heartbeat property, stack trace"
+               + "is: %s", ExceptionUtils.getStackTrace(e));
+          }
+        }
+        return hasSetValues;
+      }
+    };
+  }
+
+  /**
+   * This method initializes the collection with Default Properties of this provider.
+   * @param defaultFields collection to hold default properties.
+   */
+  private void initializeDefaultFields(Set<String> defaultFields) {
+
+    if (defaultFields == null) {
+      defaultFields = new HashSet<>();
+    }
+    defaultFields.add("jdkVersion");
+    defaultFields.add("sdk-version");
+    defaultFields.add("osType");
+    defaultFields.add("processSessionId");
+  }
+
+  /**
+   * Gets the JDK version being used by the application
+   * @return String representing JDK Version
+   */
+  private String getJdkVersion() {
```

Comment:
It's not the _JdkVersion_ it's the _JreVersion_

Reply:
Fixed